packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1525,6 +1524,34 @@ export class ContainerRuntime
 			opGroupingManager,
 		);
 
+		const deltaManagerSummarizerProxy = new DeltaManagerSummarizerProxy(this.innerDeltaManager);
+		const pendingRuntimeState = pendingLocalState as IPendingRuntimeState | undefined;
+		this.pendingStateManager = new PendingStateManager(
+			{
+				applyStashedOp: this.applyStashedOp.bind(this),
+				clientId: () => this.clientId,
+				close: this.closeFn,
+				connected: () => this.connected,
+				reSubmit: (message: IPendingBatchMessage) => {
+					this.reSubmit(message);
+					this.flush();
+				},
+				reSubmitBatch: this.reSubmitBatch.bind(this),
+				isActiveConnection: () => this.innerDeltaManager.active,
+				isAttached: () => this.attachState !== AttachState.Detached,
+			},
+			pendingRuntimeState?.pending,
+			this.logger,
+		);
+
+		const useDeltaManagerOpsProxy =
+			this.mc.config.getBoolean("Fluid.ContainerRuntime.DeltaManagerOpsProxy") === true;
```

Comment:
I'm much bigger fan of kill-bit switches, as opposed to flights (i.e. On by default).
Reason for that - I want to see the existing pipeline (unit tests, E2E tests, service tests, tests in Loop, etc.) testing what we are shipping. When we flight things in prod, we skip all of that, which means we are finding bugs in production as opposed to much earlier.

Same can be said about old state (when we use kill-bit and stop testing old stuff), but if code is written right, it should bring us to well-known state that was already tested that way. No guarantees on the future (system deteriorating and thus old mode stop working after some time), but I think the balance is still better with kill-bit direction.

There is a place for flighting, but in most cases I think we should make our life simpler and less risky.

Reply:
Changed to kill bit